library/core/src/main/java/com/google/android/exoplayer2/source/SampleQueue.java

```diff
@@ -673,6 +676,23 @@ private synchronized void commitSample(
     largestQueuedTimestampUs = Math.max(largestQueuedTimestampUs, timeUs);
 
     int relativeEndIndex = getRelativeIndex(length);
+
+    // Check for obvious discontinuities in the content
+    if ( length > 0 && (sampleFlags & C.BUFFER_FLAG_KEY_FRAME) == 1) {
```

Comment:
How is this change going to work with subtitle and metadata tracks, which are often quite sparse? A 10 second gap is not unexpected, and probably occurs quite regularly.

In general, I worry that this kind of change solves one problem (for media that's actually broken) whilst risking introducing new problems (for media that's actually valid). It's not even obviously wrong for a video track to have a >10 second gap between frames, for some more unusual cases. For example if a video track is being used for album artwork.

Reply:
We have sample streams from MusicChoice which we support that we can try, but I'm not a fan of hard-coded values.  The goal is to detect intra-Extractor deviations, so we could check against and average value, this dos need more thought.